pkg/collector/corechecks/containerlifecycle/check.go

```diff
@@ -0,0 +1,139 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package containerlifecycle
+
+import (
+	"errors"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	ddConfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+const (
+	checkName           = "container_lifecycle"
+	maxChunkSize        = 100
+	defaultPollInterval = 10
+)
+
+// Config holds the container_lifecycle check configuration
+type Config struct {
+	chunkSize    int `yaml:"chunk_size"`
+	pollInterval int `yaml:"poll_interval_seconds"`
+}
+
+// Parse parses the container_lifecycle check config and set default values
+func (c *Config) Parse(data []byte) error {
+	return yaml.Unmarshal(data, c)
+}
+
+// Check reports container lifecycle events
+type Check struct {
+	core.CheckBase
+	workloadmetaStore workloadmeta.Store
+	instance          *Config
+	processor         *processor
+	stopCh            chan struct{}
+}
+
+// Configure parses the check configuration and initializes the container_lifecycle check
+func (c *Check) Configure(config, initConfig integration.Data, source string) error {
+	if !ddConfig.Datadog.GetBool("container_lifecycle.enabled") {
+		return errors.New("collection of container lifecycle events is disabled")
+	}
+
+	var err error
+
+	err = c.CommonConfigure(config, source)
+	if err != nil {
+		return err
+	}
+
+	err = c.instance.Parse(config)
+	if err != nil {
+		return err
+	}
+
+	sender, err := aggregator.GetSender(c.ID())
+	if err != nil {
+		return err
+	}
+
+	if c.instance.chunkSize <= 0 || c.instance.chunkSize > maxChunkSize {
+		c.instance.chunkSize = maxChunkSize
```

Comment:
Just for confirmation: the default chunkSize correspond to the Max?

Reply:
yes